.github/workflows/run_all_frameworks.yml

```diff
@@ -0,0 +1,68 @@
+name: Run All Frameworks
+
+on:
+  pull_request:
+    paths:
+      - 'amlb/**'
+      - 'frameworks/shared/**'
+      - '.github/workflows/run_all_frameworks.yml'
+      - '.github/runbenchmark/action.yml'
+      - 'runbenchmark.py'
+      - 'requirements.txt'
+
+
+jobs:
+  baseline:
+    name: Run Baseline on OpenML Iris
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: pip-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+            pip-
+      - name: Setup Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+      - name: Install Requirements
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+      - name: Run constantpredictor on openml iris
+        run: |
+          python runbenchmark.py constantpredictor openml/t/59 1h4c
+      - name: Echo Python Files
+        run: |
+          python -c "import numpy; print(numpy.__file__)"
+
+  run_frameworks:
+    name: Run validation on all frameworks
+    runs-on: ubuntu-latest
+    needs: baseline
+    strategy:
+      matrix:
+        python-version: [3.8]
+        framework: [autoxgboost, autogluon, gama, h2oautoml, mlplan, tpot, autosklearn]
+        task: [bioresponse, dresses-sales, eucalyptus, internet-advertisements, micro-mass, kc1, APSFailure]
```

Comment:
do we really need all those tasks? also some of those datasets are not small and I don't want to have to wait for the validation workflow for several hours after each commit: test constraint is 10min per task.
If we keep 7 datasets, that we run with 10 frameworks, it makes a  lot of computation. I guess it runs in parallel, but are we not limited in hardware usage?

Reply:
As far as I could tell, we are not limited (other than max 256 parallel jobs). I haven't ran this on all frameworks yet (some still break). During testing I found that quite often most tasks were started in parallel, so in practice it might not take longer than ~20 minutes for the workflow.
I would stay on the side of thoroughness, we should expand our regular unit test suite for quick tests with good coverage. This workflow is then only to make sure we didn't miss anything before merging with `master`, not for regular debugging.